lib/sdm/maven.ts

```diff
@@ -0,0 +1,235 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitHubRepoRef } from "@atomist/automation-client";
+import { scanFreePort } from "@atomist/automation-client/lib/util/port";
+import {
+    actionableButton,
+    and,
+    CommandHandlerRegistration,
+    DoNotSetAnyGoals,
+    execPromise,
+    GeneratorRegistration,
+    goal,
+    hasFile,
+    hasFileWithExtension,
+    not,
+    or,
+    SdmGoalState,
+    slackSuccessMessage,
+} from "@atomist/sdm";
+import {
+    configure,
+    Version,
+} from "@atomist/sdm-core";
+import {
+    dotnetCoreBuilder,
+    DotnetCoreProjectFileCodeTransform,
+    DotnetCoreProjectVersioner,
+    DotnetCoreVersionProjectListener,
+    getDockerfile,
+} from "@atomist/sdm-pack-analysis-dotnet";
+import { Build } from "@atomist/sdm-pack-build";
+import {
+    DockerBuild,
+    HasDockerfile,
+} from "@atomist/sdm-pack-docker";
+import {
+    mavenBuilder,
+    MavenProjectVersioner,
+    MvnVersion,
+    SpringProjectCreationParameterDefinitions,
+    SpringProjectCreationParameters,
+    TransformMavenSpringBootSeedToCustomProject,
+} from "@atomist/sdm-pack-spring";
+import {
+    bold,
+    codeLine,
+    url,
+} from "@atomist/slack-messages";
+import { replaceSeedSlug } from "../transform/replaceSeedSlug";
+import { UpdateReadmeTitle } from "../transform/updateReadmeTitle";
+
+/**
+ * Atomist SDM Sample
+ * @description SDM to create and build Maven projects
+ * @tag sdm,generator,maven
+ * @instructions <p>Now that the SDM is up and running, create a new .NET Core
+ *               project by running '@atomist create maven project' and
+ *               observe how the SDM will build and dockerize the new project.
+ *
+ *               The docker build and run goals require a locally accessible
+ *               docker daemon. Please make sure to configure your terminal for
+ *               docker access.</p>
+ */
+
+// atomist:code-snippet:start=mavenGenerator
+/**
+ * .NET Core generator registration
+ */
+const MavenGenerator: GeneratorRegistration<SpringProjectCreationParameters> = {
+    name: "MavenGenerator",
+    intent: "create maven project",
+    description: "Creates a new Maven project",
+    tags: ["dotnet"],
+    autoSubmit: true,
+    parameters: SpringProjectCreationParameterDefinitions,
+    startingPoint: GitHubRepoRef.from({ owner: "atomist-seeds", repo: "spring-rest", branch: "master" }),
+    transform: [
+        UpdateReadmeTitle,
+        ...TransformMavenSpringBootSeedToCustomProject,
+    ],
+};
+// atomist:code-snippet:end
+
+/**
+ * Command to stop a container by provided container id
+ */
+const StopDockerContainerCommand: CommandHandlerRegistration<{ containerId: string }> = {
```

Comment:
Should we extract this command into a separate sample and reference it from there?

Reply:
It's no longer being used in the Maven sample. However, I would like to keep the examples as self-contained as possible so that people only need to look at one file.